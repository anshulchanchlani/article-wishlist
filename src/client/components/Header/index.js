import React, { Component } from 'react';
import {NavLink} from 'react-router-dom';
import {connect} from 'react-redux';
import { loginUser } from '../../actions/authActions'


import './index.scss'
class Header extends Component {
    constructor(props) {
        super(props);
        this.state = {};
    }
    componentDidMount(){
        var elem = document.querySelector(".sidenav");
        var instance = M.Sidenav.init(elem, {
            edge: "left",
            inDuration: 250
        });
    }
    renderAuthButton(){
        return  this.props.auth? <NavLink to="/logout">Logout</NavLink>:<NavLink to="/login">Login</NavLink>
    }
    renderWishlistButton(){

        return this.props.auth?<li><NavLink to="/wishlist">Wishlist  {this.props.wishlist.length}</NavLink></li> :null
    }
    render() {
        return (
            <div>
            <nav className="nav-extended">
          
           
            
            <div className="nav-wrapper black ">
            <NavLink className="brand-logo custom-brand-style" to="/">Brand</NavLink>
            <a href="#" data-target="mobile-nav" className="sidenav-trigger"><i className="material-icons">menu</i></a>
              <ul id="nav-mobile" className="right hide-on-med-and-down">
                
                {this.props.auth?<li><NavLink to="/">Items</NavLink></li>:null}
                {this.props.wishlist?this.renderWishlistButton():null}
                <li>{this.renderAuthButton()}</li>
              </ul>
            </div>
           
          </nav>
         <ul className="sidenav" id="mobile-nav">
           
           {this.props.auth?<li><NavLink to="/">Items</NavLink></li>:null}
           {this.props.wishlist?this.renderWishlistButton():null}
           <li>{this.renderAuthButton()}</li>
         </ul>
         </div>



 


        );
    }
}

function mapStateToProps(state){

    return {auth:state.auth,wishlist:state.wishlist}
}

export default connect(mapStateToProps)(Header);