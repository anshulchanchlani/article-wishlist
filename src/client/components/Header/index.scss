.custom-brand-style{
    margin-left:20px;
}