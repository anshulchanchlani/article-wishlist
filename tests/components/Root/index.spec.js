import React from 'react';
import {mockStore} from '../../setup/mockStore'
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Header from '../../../src/client/components/Header'
import Routes from '../../../src/client/routes'
const store = mockStore()
describe('Root renders',()=>{
    it('Root Component renders correctly',()=>{
        const wrapper = shallow(<Provider store={store}>
            <BrowserRouter>
                <div>
                    <Header/>
                    <Routes />
                </div>
            </BrowserRouter>
        </Provider>); 
        const jsonWrapper = toJson(wrapper)
        
        expect(jsonWrapper).toMatchSnapshot();
        
    })
})