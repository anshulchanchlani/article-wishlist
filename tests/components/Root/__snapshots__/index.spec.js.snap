// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root renders Root Component renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {},
    }
  }
>
  <BrowserRouter>
    <div>
      <Connect(Header) />
      <Routes />
    </div>
  </BrowserRouter>
</ContextProvider>
`;
